uefi/src/proto/device_path/mod.rs

```diff
@@ -975,4 +988,35 @@ mod tests {
         let owned_dp_ref = &*owned_dp;
         assert_eq!(owned_dp_ref, dp)
     }
+
+    #[test]
+    fn test_device_path_node_from_bytes() {
+        let mut raw_data = Vec::new();
+        let node = [0xa0, 0xb0];
+        let node_data = &[10, 11];
+
+        // Raw data is less than size of a [`DevicePathNode`].
+        raw_data.push(node[0]);
+        assert!(<&DevicePathNode>::try_from(raw_data.as_slice()).is_err());
+
+        // Raw data is long enough to hold a [`DevicePathNode`].
+        raw_data.push(node[1]);
+        raw_data.extend(
+            u16::try_from(mem::size_of::<DevicePathHeader>() + node_data.len())
+                .unwrap()
+                .to_le_bytes(),
+        );
+        raw_data.extend(node_data);
+        let dp = <&DevicePathNode>::try_from(raw_data.as_slice());
+        assert!(dp.is_ok());
```

Comment:
nit: just unwrap the value on the previous line, that's functionally the same as asserting `is_ok`

Reply:
my thinking was that the implicit panic would be less clear than then explicit assertion, but I could see it either way.